File: src/modifiers/blur.ts
import { createModifier } from './createModifier';

export interface BlurParams {
  /// Gaussian blur radius in points. Larger values blur more.
  radius: number;
  /// When `true`, the blur is opaque (fully obscures content behind the
  /// view's bounds); when `false` (default) the blur is transparent.
  opaque?: boolean;
}

/// SwiftUI `.blur(radius:opaque:)`. Applies a Gaussian blur to the view.
export function blur(params: BlurParams): ReturnType<typeof createModifier>;
export function blur(radius: number): ReturnType<typeof createModifier>;
export function blur(a: BlurParams | number) {
  if (typeof a === 'number') return createModifier('blur', { radius: a });
  return createModifier('blur', { opaque: false, ...a });
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/Blur.swift
import SwiftUI

/// SwiftUI `.blur(radius:opaque:)`. Applies a Gaussian blur to the view.
enum RNWBlurModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("blur") { view, params, _ in
            AnyView(view.blur(
                radius: params.cgFloat("radius") ?? 0,
                opaque: params.bool("opaque") ?? false
            ))
        }
    }
}
